TEIIDTOOLS-419: Ports the odata query editor from DSB

phantomjinx opened the pull request:
* package.json
 * Needs lodash because well its lodash!
 * x2js for parsing xml into json

* dataservice-card
* datasevice-list
 * Provides an odata icon to the card if a service is published allowing
   for opening the odata query editor.

* dataservices-compoment
 * Show the odata editor

* .../odata-control/...
 * Component UI for the odata editor

* virt-route
 * Model for a virtualization's routes of the published service, eg. odata

phantomjinx committed:
TEIIDTOOLS-419: Ports the odata query editor from DSB

* package.json
 * Needs lodash because well its lodash!
 * x2js for parsing xml into json

* dataservice-card
* datasevice-list
 * Provides an odata icon to the card if a service is published allowing
   for opening the odata query editor.

* dataservices-compoment
 * Show the odata editor

* .../odata-control/...
 * Component UI for the odata editor

* virt-route
 * Model for a virtualization's routes of the published service, eg. odata

phantomjinx committed:
TEIIDTOOLS-400: Improve asynchronous performance of publishing

* Adds and tracks extra states of submitting and configuring

* Better able to understand how the publishing is going rather than seeing
  a single spinning icon that may have gone wrong.

phantomjinx committed:
Updated publishing status icons

* Changes the icons to be different rather than just the colour.

elvisisking commented:
I'm getting this error when I run `mvn clean install`:

`ERROR in ng:///Users/dan/Temp/pj-odata/target/copy-src/src/app/dataservices/odata-control/odata-control.component.html (71,11): Property 'odata' is private and only accessible within class 'OdataControlComponent'.`

If I fix that error in `odata-control.component.ts` I then get this:

`ERROR [karma]: Error: /Users/dan/Temp/pj-odata/target/copy-src/src/app/dataservices/odata-control/odata-control.component.spec.ts (17,37): Cannot find module './sql-control.component'.,Module not found: Error: Can't resolve './sql-control.component' in '/Users/dan/Temp/pj-odata/target/copy-src/src/app/dataservices/odata-control'`

This error can be fixed by changing the import to this:

`import { SqlControlComponent } from "@dataservices/sql-control/sql-control.component";`

However, when that error is fixed I still get 6 test failures. Can you get the build to run @phantomjinx ?

phantomjinx committed:
Fix API incompatibility of patternfly-ng

* Patternfly-ng appears to have been updated and the new version causes a
  builds error due to a lack of PatternFlyNgModule in the library.

* For the moment, fix the dependency version to 3.1.5.

phantomjinx committed:
TEIIDTOOLS-436: Convert the odata results to json and xml

* Adds icons for odata controls panels for the opening/closing the
  accordion groups

* Small changes and fixes to layout

* Adds spinner during initial metadata fetch and stop error message
  displaying prematurely

* Uses CodeMirror for odata results

* Adds Output Format control to allow choosing between XML and JSON

* Uses ViewChild to access the CodeMirror instance, allowing changing of
  the mode between XML and JSON

* DataserviceService
 * Ensures return of odata results is a string and in turn adds that string
   as a value to a json object.
 * Ensure the XML is formatted/beautified before its return

* Specialise other references to the .CodeMirror class to ensure they do
  not conflict with one another.

phantomjinx committed:
Fixes the maven build

* pom.xml
 * Modify the phases to ensure npm gets run first to generate the
   node_modules directory and all the dependencies.
 * Avoid copying any generated node_modules directory from the src which
   may not have yet been created (if a manual build has never been run)

* package.json
 * osenv seems to be required by maven.

* dataservice.component.spec.ts
 * Fixes test failures by ensuring dependencies of OdataControlComponent

phantomjinx commented:
@elvisisking 

Few more commits added to the PR related to odata-editor. Latter one fixes the tests and maven now works ok.

phantomjinx committed:
Turns off unnecessary lint rules and fixes remaining lint errors

elvisisking reviewed (changes requested):
Fix the max-length rule like we talked about then this is ready to merge. Nice work @phantomjinx.
